### ISSUE TITLE
Improvement of the "Filter" feature of the Shortcuts Mapper

### ISSUE THREAD

**Issue by guy038** [notepad-plus-plus/notepad-plus-plus#9316]
Hi, **All**,

Presently, the **`Filter`** functionality, of the ***Shortcut Mapper** acts on the **`Command Name**`column, only. Could it be possible to consider **filtering** on the **`Shortcut`** column, too ?

Buy this way, it would be easier to choose an **appropriate** shortcut, entering the **`+`** symbol first !

---

Some examples :

Searching **`+W`** would display

- In the **`Main menu`** tab :

~~~z
|    1  |  Close                                |  Ctrl+W                |   File   |
|    2  |  Close All                            |  Ctrl+Shift+W          |   File   |
~~~

And searching **`+1`** would display

- In the **`Main menu`** tab :

~~~z
|    1  |  Previous mark using 1st style        |  Ctrl+Shift+1          |  Search  |
|    2  |  Next mark using 1st style            |  Ctrl+1                |  Search  |
|    3  |  Collapse Level 1                     |  Alt+1                 |   View   |
|    4  |  Uncollapse Level 1                   |  Alt+Shift+1           |   View   |
~~~

---

Of course, sometimes, some **additional** entries would be output, as well, because the **key** name begins with that letter . But this is really a **minor** drawback !

For instance, searching for **`+P`** would display

- In the **`Main menu`** tab :

~~~z
|    1  |  Print...                             |  Ctrl+P                |   File   |
|    2  |  Next Tab                             |  Ctrl+Page down        |   View   |
|    3  |  Previous Tab                         |  Ctrl+Page up          |   View   |
|    4  |  Move Tab Forward                     |  Ctrl+Shift+Page down  |   View   |
|    5  |  Move Tab Backward                    |  Ctrl+Shift+Page up    |   View   |
|    6  |  Playback                             |  Ctrl+Shift+P          |   Edit   |
~~~

- In the **`Scintilla commands`** tab :

~~~z
|    1  |  SCI_PAGEUPEXTEND                     |  Shift+Page up         |
|    2  |  SCI_PAGEUPRECTEXTEND                 |  Alt+Shift+Page up     |
|    3  |  SCI_PAGEDOWNEXTEND                   |  Shift+Page down       |
|    4  |  SCI_PAGEDOWNRECTEXTEND               |  Alt+Shift+Page down   |
~~~

Now, searching for **`+U`** would inform the user that **no more** shortcut, with letter **`U`**, are **available** !

- In **`Main menu`** tab

~~~z
|    1  |  UPPERCASE                            |  Ctrl+Shift+U          |   Edit   |
|    2  |  lowercase                            |  Ctrl+U                |   Edit   |
|    3  |  Proper Case                          |  Alt+U                 |   Edit   |
|    4  |  Proper Case (blend)                  |  Alt+Shift+U           |   Edit   |
|    5  |  Sentence case                        |  Ctrl+Alt+U            |   Edit   |
|    6  |  Sentence case (blend)                |  Ctrl+Alt+Shift+U      |   Edit   |
|    7  |  Move Up Current Line                 |  Ctrl+Shift+Up         |   Edit   |
~~~

- In the **`Scintilla commands`** tab :

~~~z
|    1  |  SCI_LINEUPEXTEND                     |  Shift+Up              |
|    2  |  SCI_LINEUPRECTEXTEND                 |  Alt+Shift+Up          |
|    3  |  SCI_LINESCROLLUP                     |  Ctrl+Up               |
~~~

---

Only **two** values **`+S`** and **`+A`** would really differ from what it's normally **expected** because of the strings **`+Shift`** and **`+Alt`**, that frequently occur in **shortcuts** !

For instance, searching for **`+A`**, would display

- In the **`Main menu`** tab :

~~~z
|    1  |  Save As...                           |  Ctrl+Alt+S            |   File   |
|    2  |  Sentence case                        |  Ctrl+Alt+U            |   Edit   |
|    3  |  Sentence case (blend)                |  Ctrl+Alt+Shift+U      |   Edit   |
|    4  |  Insert Blank Line Above Current      |  Ctrl+Alt+Enter        |   Edit   |
|    5  |  Insert Blank Line Below Current      |  Ctrl+Alt+Shift+Enter  |   Edit   |
|    6  |  Path Completion                      |  Ctrl+Alt+Spacebar     |   Edit   |
|    7  |  Find (Volatile) Next                 |  Ctrl+Alt+F3           |  Search  |
|    8  |  Find (Volatile) Previous             |  Ctrl+Alt+Shift+F3     |  Search  |
|    9  |  Incremental Search                   |  Ctrl+Alt+I            |  Search  |
|   10  |  Select All Between Matching Braces   |  Ctrl+Alt+B            |  Search  |
|   11  |  Collapse Current Level               |  Ctrl+Alt+F            |   View   |
|   12  |  Uncollapse Current Level             |  Ctrl+Alt+Shift+F      |   View   |
|   13  |  Text Direction RTL                   |  Ctrl+Alt+R            |   Edit   |
|   14  |  Text Direction LTR                   |  Ctrl+Alt+L            |   Edit   |
~~~

- In the **`Scintilla commands`** tab :

~~~z
|    1  |  SCI_SELECTALL                        |  Ctrl+A                |          |
~~~

And searching for **+S`** would display

- In the **`Main menu`** tab :

~~~z
|    1  |  Save                                 |  Ctrl+S                |   File   |
|    2  |  Save As...                           |  Ctrl+Alt+S            |   File   |
|    3  |  Save All                             |  Ctrl+Shift+S          |   File   |
|    4  |  Close All                            |  Ctrl+Shift+W          |   File   |
|    5  |  Restore Recent Closed File           |  Ctrl+Shift+T          |   File   |
|    6  |  UPPERCASE                            |  Ctrl+Shift+U          |   Edit   |
|    7  |  Proper Case (blend)                  |  Alt+Shift+U           |   Edit   |
|    8  |  Sentence case (blend)                |  Ctrl+Alt+Shift+U      |   Edit   |
|    9  |  Move Up Current Line                 |  Ctrl+Shift+Up         |   Edit   |
|   10  |  Move Down Current Line               |  Ctrl+Shift+Down       |   Edit   |
|   11  |  Insert Blank Line Below Current      |  Ctrl+Alt+Shift+Enter  |   Edit   |
|   12  |  Single Line Uncomment                |  Ctrl+Shift+K          |   Edit   |
|   13  |  Block Comment                        |  Ctrl+Shift+Q          |   Edit   |
|   14  |  Function Completion                  |  Ctrl+Spacebar         |   Edit   |
|   15  |  Path Completion                      |  Ctrl+Alt+Spacebar     |   Edit   |
|   16  |  Function Parameters Hint             |  Ctrl+Shift+Spacebar   |   Edit   |
|   17  |  Find in Files...                     |  Ctrl+Shift+F          |  Search  |
|   18  |  Select and Find Previous             |  Ctrl+Shift+F3         |  Search  |
|   19  |  Find (Volatile) Previous             |  Ctrl+Alt+Shift+F3     |  Search  |
|   20  |  Previous mark using 1st style        |  Ctrl+Shift+1          |  Search  |
|   21  |  Previous mark using 2nd style        |  Ctrl+Shift+2          |  Search  |
|   22  |  Previous mark using 3rd style        |  Ctrl+Shift+3          |  Search  |
|   23  |  Previous mark using 4th style        |  Ctrl+Shift+4          |  Search  |
|   24  |  Previous mark using 5th style        |  Ctrl+Shift+5          |  Search  |
|   25  |  Previous mark created with Mark      |  Ctrl+Shift+0          |  Search  |
|   26  |  Move Tab Forward                     |  Ctrl+Shift+Page down  |   View   |
|   27  |  Move Tab Backward                    |  Ctrl+Shift+Page up    |   View   |
|   28  |  Switch to previous document          |  Ctrl+Shift+Tab        |   View   |
|   29  |  Unfold All                           |  Alt+Shift+0           |   View   |
|   30  |  Uncollapse Current Level             |  Ctrl+Alt+Shift+F      |   View   |
|   31  |  Uncollapse Level 1                   |  Alt+Shift+1           |   View   |
|   32  |  Uncollapse Level 2                   |  Alt+Shift+2           |   View   |
|   33  |  Uncollapse Level 3                   |  Alt+Shift+3           |   View   |
|   34  |  Uncollapse Level 4                   |  Alt+Shift+4           |   View   |
|   35  |  Uncollapse Level 5                   |  Alt+Shift+5           |   View   |
|   36  |  Uncollapse Level 6                   |  Alt+Shift+6           |   View   |
|   37  |  Uncollapse Level 7                   |  Alt+Shift+7           |   View   |
|   38  |  Uncollapse Level 8                   |  Alt+Shift+8           |   View   |
|   39  |  Toggle macro recording               |  Ctrl+Shift+R          |   Edit   |
|   40  |  Playback                             |  Ctrl+Shift+P          |   Edit   |
~~~

- In the **`Macros`** tab

~~~z
|    1  |  Trim Trailing Space and Save         |  Alt+Shift+S           |
~~~

- In the **`Scintilla commands`** tab :

~~~z 
|    1  |  SCI_REDO                             |  Ctrl+Shift+Z          |
|    2  |  SCI_LINEDOWNRECTEXTEND               |  Alt+Shift+Down        |
|    3  |  SCI_LINEUPRECTEXTEND                 |  Alt+Shift+Up          |
|    4  |  SCI_PARADOWNEXTEND                   |  Ctrl+Shift+]          |
|    5  |  SCI_PARAUPEXTEND                     |  Ctrl+Shift+[          |
|    6  |  SCI_CHARLEFTRECTEXTEND               |  Alt+Shift+Left        |
|    7  |  SCI_CHARRIGHTRECTEXTEND              |  Alt+Shift+Right       |
|    8  |  SCI_WORDLEFTEXTEND                   |  Ctrl+Shift+Left       |
|    9  |  SCI_WORDRIGHTENDEXTEND               |  Ctrl+Shift+Right      |
|   10  |  SCI_WORDPARTLEFTEXTEND               |  Ctrl+Shift+/          |
|   11  |  SCI_WORDPARTRIGHTEXTEND              |  Ctrl+Shift+\          |
|   12  |  SCI_VCHOMERECTEXTEND                 |  Alt+Shift+Home        |
|   13  |  SCI_LINEENDRECTEXTEND                |  Alt+Shift+End         |
|   14  |  SCI_DOCUMENTSTARTEXTEND              |  Ctrl+Shift+Home       |
|   15  |  SCI_DOCUMENTENDEXTEND                |  Ctrl+Shift+End        |
|   16  |  SCI_PAGEUPRECTEXTEND                 |  Alt+Shift+Page up     |
|   17  |  SCI_PAGEDOWNRECTEXTEND               |  Alt+Shift+Page down   |
|   18  |  SCI_DELLINELEFT                      |  Ctrl+Shift+Backspace  |
|   19  |  SCI_DELLINERIGHT                     |  Ctrl+Shift+DEL        |
|   20  |  SCI_LINEDELETE                       |  Ctrl+Shift+L          |
|   21  |  SCI_LINECOPY                         |  Ctrl+Shift+X          |
~~~

In short, despite some **approximative** results, I do think that the returned information would be **valuable** !

Best Regards,

guy038

**Comment by ArkadiuszMichalski** [comment]
Duplicate of https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5616.
